### DISCUSSION TITLE
Is there a way to append a JSON object to a jsonb array in a Supabase database without fetching and updating it separately?

### DISCUSSION THREAD

**Discussion by udayalawa** [supabase/supabase#18829] [Questions] [answered]
How can I append a new JSON object to a jsonb array in the database without fetching and updating it separately? For example, if I have a testcolumn in the testdb table with the current value [{"id":1,"name":"shah"}], how can I push the value [{"id":2,"name":"rukh"}] to it directly in a single operation without fetching and updating?

any language will work, PS: Curl++

**Comment by GaryAustin1** [comment, marked as answer]
Not thru the API.  You can't even add more value to a json object or push to a basic array.
You will need to use an rpc call to a Postgres function and do the operation in SQL.

**Reply by udayalawa** [reply]
Thanks for reply, 
by using rpc i was able to do all sort of stuff. 

thanks again.

**Reply by lorikku** [reply]
@udayalawa could you provide your RPC solution so others can also benefit? 😄

**Reply by udayalawa** [reply]
> @udayalawa could you provide your RPC solution so others can also benefit? 😄 



It's Been a year but as far as I remember I think you have to play with value.OLD and value.NEW (database triggers before update) and append from there
Or create an rpc that will expect append (as parameter) value then it will fetch current value to append and update it with input.

I'll give you a dummy code quite occupied at the moment.
